src/OlStyler.js

```diff
@@ -1,525 +1,33 @@
-              ? hexToRGB(fill.fill, fill.fillOpacity)
-              : fill.fill,
-        }),
-      }),
-    });
-  }
-  return new Style({});
-}
-
-// Create memoized versions of the style converters.
-// They use a WeakMap to return the same OL style object if the symbolizer is the same.
-// Note: this only works for constant symbolizers!
-// Todo: find a smart way to optimize text symbolizers.
-
-/**
- * Function to memoize style conversion functions that convert sld symbolizers to OpenLayers style instances.
- * The memoized version of the style converter returns the same OL style instance if the symbolizer is the same object.
- * Uses a WeakMap internally.
- * Note: This only works for constant symbolizers.
- * Note: Text symbolizers depend on the feature property and the geometry type, these cannot be cached in this way.
- * @private
- * @param {Function} styleFunction Function that accepts a single symbolizer object and returns the corresponding OpenLayers style object.
- * @returns {Function} The memoized function of the style conversion function.
- */
-function memoize(styleFunction) {
-  const styleCache = new WeakMap();
-
-  return symbolizer => {
-    let olStyle = styleCache.get(symbolizer);
-
-    if (!olStyle) {
-      olStyle = styleFunction(symbolizer);
-      styleCache.set(symbolizer, olStyle);
-    }
-
-    return olStyle;
-  };
-}
-
-// Memoized versions of point, line and polygon style converters.
-const cachedPointStyle = memoize(pointStyle);
-const cachedLineStyle = memoize(lineStyle);
-const cachedPolygonStyle = memoize(polygonStyle);
+import {
+  getCachedImageUrls,
+  processExternalGraphicSymbolizers,
+} from './imageCache';
+import { defaultPointStyle } from './styles/static';
+import getPointStyle from './styles/pointStyle';
+import getLineStyle from './styles/lineStyle';
+import getPolygonStyle from './styles/polygonStyle';
+import getTextStyle from './styles/textStyle';
 
 const defaultStyles = [defaultPointStyle];
 
-
 /**
- * Pushes style to array
- * @example pushTo(styles, point[j], cachedPointStyle);
- * @param {array} styles rulesconverter
- * @param {object} item style description of array of styles description
- * @param {function} cache function that returns cached style
+ * Convert symbolizers together with the feature to OL style objects and append them to the styles array.
+ * @example appendStyle(styles, point[j], feature, getPointStyle);
+ * @param {Array<ol/style>} styles Array of OL styles.
+ * @param {object|Array<object>} symbolizers Feature symbolizer object, or array of feature symbolizers.
+ * @param {ol/feature} feature OpenLayers feature.
+ * @param {Function} styleFunction Function for getting the OL style object. Signature (symbolizer, feature) => OL style.
  */
-function pushTo(array, item, cache) {
-  if (Array.isArray(item)) {
-    for (let k = 0; k < item.length; k += 1) {
-      array.push(cache(item[k]));
+function appendStyle(styles, symbolizers, feature, styleFunction) {
```

Comment:
I think: symbolyzers shouldn't be of mixed type (see code from previous pull request)

Reply:
I agree, but that's a refectoring worth its own branch.